controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -323,3 +358,144 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error
 		).
 		Complete(r)
 }
+
+// ------------------------------synthetic-monitoring--------------------------
+
+func (r *HostedControlPlaneReconciler) getDynatraceSecrets(ctx context.Context) (string, string, error) {
+
+	secret := &v1.Secret{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: secretNamespace}, secret)
+	if err != nil {
+		return "", "", fmt.Errorf("error getting Kubernetes secret: %v", err)
+	}
+
+	valueBytesDynatraceApiToken, ok := secret.Data[dynatraceApiKey]
+	if !ok {
+		return "", "", fmt.Errorf("secret did not contain key %s", dynatraceApiKey)
+	}
+	if len(valueBytesDynatraceApiToken) == 0 {
+		return "", "", fmt.Errorf("%s is empty", dynatraceApiKey)
+	}
+	valueDynatraceApiToken := string(valueBytesDynatraceApiToken)
+
+	valueBytesDynatraceTenant, ok := secret.Data[dynatraceTenantKey]
+	if !ok {
+		return "", "", fmt.Errorf("secret did not contain key %s", dynatraceTenantKey)
+	}
+	if len(valueBytesDynatraceTenant) == 0 {
+		return "", "", fmt.Errorf("%s is empty", dynatraceTenantKey)
+	}
+	valueDynatraceTenant := string(valueBytesDynatraceTenant)
+
+	return valueDynatraceApiToken, valueDynatraceTenant, nil
+}
+
+func getDynatraceHTTPMonitorID(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, error) {
+	labels := hostedcontrolplane.GetLabels()
+	dynatraceHttpMonitorId, exists := labels[httpMonitorLabel]
+	if exists {
+		return dynatraceHttpMonitorId, nil
+	}
+	return "", fmt.Errorf("key '%s' not found in labels", httpMonitorLabel)
+}
```

Comment:
```suggestion
func getDynatraceHTTPMonitorID(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, bool) {
	labels := hostedcontrolplane.GetLabels()
	return labels[httpMonitorLabel]
}
```
This isn't really an error if it doesn't exist based on how this function is used